packages/bubbles-map/src/MapBubble.tsx

```diff
@@ -0,0 +1,139 @@
+    if (cols.length === 3) {
+      return (
+        (cols[0].type === "TEXT" || cols[0].type === "LABEL") &&
+        cols[1].type === "NUMBER" &&
+        (cols[2].type === "TEXT" || cols[2].type === "LABEL")
+      );
+    }
+    // Case where lat and long are provided in different columns
+    if (cols.length === 4) {
+      return (
+        (cols[0].type === "TEXT" || cols[0].type === "LABEL") &&
+        cols[1].type === "NUMBER" &&
+        cols[2].type === "NUMBER" &&
+        cols[3].type === "NUMBER"
+      );
+    }
+  }
+
+  return false;
+}
+
+interface Props {
+  controller: ComponentController;
+}
+
+export function MapBubble(props: Props) {
+  const [mapProps, setMapProps] = useState<LatLongProps>({
+    title: DEFAULT_TITLE,
+    latitude: DEFAULT_LATITUDE,
+    longitude: DEFAULT_LONGITUDE,
+    titleenabled: DEFAULT_TITLE_ENABLED,
+    zoom: DEFAULT_ZOOM,
+    bubblecolor: DEFAULT_COLOR,
+    maxRadius: DEFAULT_MAX_RADIUS,
+    minRadius: DEFAULT_MIN_RADIUS,
+    resizeBubbles: DEFAULT_RESIZE_BUBBLES,
+    data: DefaultData
+  });
+  const [dataset, setDataset] = useState<DataSet>();
+  let data: Data[] = DefaultData;
+  useEffect(() => {
+    props.controller.setOnInit(componentProps => {
+      setMapProps({
+        title: (componentProps.get(TITLE_PROP) as string) || DEFAULT_TITLE,
+        latitude: +(componentProps.get(LATITUDE_PROP)) || DEFAULT_LATITUDE,
+        longitude: +(componentProps.get(LONGITUDE_PROP)) || DEFAULT_LONGITUDE,
+        titleenabled: componentProps.get(TITLE_ENABLED_PROP) as boolean || DEFAULT_TITLE_ENABLED,
+        zoom: +(componentProps.get(ZOOM_PROP)) || DEFAULT_ZOOM ,
+        bubblecolor:componentProps.get(BUBBLECOLOR_PROP) as string || DEFAULT_COLOR,
+        maxRadius: +(componentProps.get(MAXRADIUS_PROP)) || DEFAULT_MAX_RADIUS,
+        minRadius: +(componentProps.get(MINRADIUS_PROP)) || DEFAULT_MIN_RADIUS,
+        resizeBubbles: componentProps.get(RESIZEBUBBLES_PROP) as boolean || DEFAULT_RESIZE_BUBBLES,
+        data: data
+      });
+    });
+    props.controller.setOnDataSet((_dataset: DataSet) => {
+      setDataset(_dataset);
+    });
+  }, [props.controller]);
+
+  let isValid = dataset ? validateDataset(dataset) : false;
+
+  // retrieving data
+  if (isValid) {
+    data = [];
+    const isLatLong = dataset && dataset.columns[2].type === "TEXT";
+
+    dataset?.data.forEach(d => {
+      let latitude: number;
+      let longitude: number;
+      if (isLatLong) {
+        const latLong = (d[2] as any) as string;
+        const latLongParts = latLong.split(",");
+        latitude = +latLongParts[0].trim();
+        longitude = +latLongParts[1].trim();
+      } else {
+        latitude = +d[2];
+        longitude = +d[3];
+      }
+
+      data.push({
+        name: (d[0] as any) as string,
+        value: +d[1],
+        latitude: latitude,
+        longitude: longitude
+      });
+    });
+  }
+
+  return (
```

Comment:
We could use the new configurationIssue method in the JS API as well - this prints a nice message on dashbuilder side. Please take a look here, what we do is we send a message to DB saying that user config is with some problem and when user fixes then we send a configuration ok

Dashbuilder receives the configuration issue and shows a standard warning

https://github.com/kiegroup/appformer/blob/master/dashbuilder/dashbuilder-shared/dashbuilder-js/packages/process-heatmap-component/src/ProcessHeatmapComponent.tsx#L136